x-pack/plugins/uptime/public/components/overview/filter_group/filter_group.test.tsx

```diff
@@ -8,67 +8,103 @@
-    ['expands filter group for Tag filter', 'Search for tag'],
-  ])('handles loading', async (popoverButtonLabel, searchInputLabel) => {
-    const { getByLabelText } = render(
-      <FilterGroupComponent loading={true} overviewFilters={overviewFilters} />
-    );
+    ['expands filter group for Location filter'],
+    ['expands filter group for Port filter'],
+    ['expands filter group for Scheme filter'],
+    ['expands filter group for Tag filter'],
+  ])('handles loading', async (popoverButtonLabel) => {
+    jest.spyOn(Hooks, 'useValuesList').mockReturnValue({
+      values: [],
+      loading: true,
+    });
+    const { getByLabelText, getByText } = render(<FilterGroup />);
 
-    const popoverButton = getByLabelText(popoverButtonLabel);
-    fireEvent.click(popoverButton);
     await waitFor(() => {
-      const searchInput = getByLabelText(searchInputLabel);
-      expect(searchInput).toHaveAttribute('placeholder', 'Loading...');
+      const popoverButton = getByLabelText(popoverButtonLabel);
+      fireEvent.click(popoverButton);
+    });
+    await waitFor(() => {
+      expect(getByText('Loading options'));
     });
   });
 
   it.each([
     [
       'expands filter group for Location filter',
-      'Search for location',
-      ['Filter by Location nyc.', 'Filter by Location fairbanks.'],
+      [
+        [
+          {
+            label: 'Fairbanks',
+            count: 10,
+          },
+          {
+            label: 'NYC',
+            count: 2,
+          },
+        ],
+        [],
+        [],
+        [],
+      ],
     ],
     [
       'expands filter group for Port filter',
-      'Search for port',
-      ['Filter by Port 5601.', 'Filter by Port 9200.'],
+      [
+        [],
+        [
+          { label: '80', count: 12 },
+          { label: '443', count: 8 },
+        ],
+        [],
+        [],
+      ],
     ],
     [
       'expands filter group for Scheme filter',
-      'Search for scheme',
-      ['Filter by Scheme http.', 'Filter by Scheme tcp.'],
+      [
+        [],
+        [],
+        [
+          { label: 'HTTP', count: 15 },
+          { label: 'TCP', count: 10 },
+        ],
+        [],
+      ],
     ],
     [
       'expands filter group for Tag filter',
-      'Search for tag',
-      ['Filter by Tag prod.', 'Filter by Tag dev.'],
+      [
+        [],
+        [],
+        [],
+        [
+          { label: 'test', count: 23 },
+          { label: 'prod', count: 10 },
+        ],
+      ],
     ],
-  ])(
-    'displays filter items when clicked',
-    async (popoverButtonLabel, searchInputLabel, filterItemButtonLabels) => {
-      const { getByLabelText } = render(
-        <FilterGroupComponent loading={false} overviewFilters={overviewFilters} />
-      );
+  ])('displays filter item counts when clicked', async (popoverButtonLabel, values) => {
+    const spy = jest.spyOn(Hooks, 'useValuesList');
+    for (let i = 0; i < 4; i++) {
+      spy.mockReturnValueOnce({
```

Comment:
I'm not sure what's happening here. Can you explain? Wondering if there's a way for us not to use the hardcoded values in the for loop.